src/poke_env/environment/abstract_battle.py

```diff
@@ -665,7 +676,10 @@ def parse_message(self, split_message: List[str]):
             self.get_pokemon(pokemon).end_effect(effect)
         elif event[1] == "-endability":
             pokemon = event[2]
-            self.get_pokemon(pokemon).ability = None
+            if len(event) > 4 and event[4].startswith("[from] move:"):
+                self.get_pokemon(pokemon).set_temporary_ability(None)
+            else:
+                self.get_pokemon(pokemon).ability = None
```

Comment:
What are examples of this path being used?

Reply:
For the if: This is tested in test_battle on line 576 (not sure how to link in the comment). `-endability` occurs whenever a `move` removes an ability, according to [showdown protocol](https://github.com/smogon/pokemon-showdown/blob/f755b2852e85715b51895ce92bb165bc58652baa/sim/pokemon.ts#L1838). It triggers first an `-endability` and then an `-ability`, as tested.

For the else: I'm not sure when `-endability` can be used to actually remove a pokemon's permanent ability. Right now, likely Gastro Acid or Neutralizing Gas triggers `-endability` without the `[from] move:` in the `event`, and this continues to be mishandled? But:
1. I'm not clear whether `-endability` should ONLY be used to set `mon._temporary_ability`
2. This is strictly less broken than before :P

If you'd like, I could do some research and circle back? If I do find that there are no permanent ways to end mons' abilities, then I can remove this `if` and create a constant within `Pokemon` (e.g. `NO_ABILITY`) to distinguish between when `Pokemon._temporary_ability` is not set and when a mon should have no ability. I'm just not confident that I can say there exists no edge-cases like this.

Reply:
Yeah let's double check - if you don't find anything we can merge this PR

Reply:
OK, looked through and I don't think there are any ways to permanently remove a mon's ability. So I took the route above and made a constant to track whether a mon is supposed to have no ability (eg removed by `Gastro Acid`) vs when we dont know one. This is abstracted away outside of this class.

Further Fuzz Tested 10K games and it seems like there were no edge-cases that I'm breaking. Also added some testing (and cleaned up syntax in `test_battle` as well) for temporary abilities